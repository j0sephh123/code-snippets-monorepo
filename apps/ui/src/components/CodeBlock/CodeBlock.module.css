/* TODO extract */
.controls {
  padding: 4px 0 0 24px;
  background-color: rgb(29, 31, 33);
  border-top: 1px solid gray;
  border-right: 1px solid gray;
  border-left: 1px solid gray;
  border-top-left-radius: 0.3em;
  border-top-right-radius: 0.3em;
}
