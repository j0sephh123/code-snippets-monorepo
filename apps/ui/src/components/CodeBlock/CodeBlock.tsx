import { ActionIcon, Box, Tooltip } from '@mantine/core';
import { Prism } from 'react-syntax-highlighter';
import { atomDark } from 'react-syntax-highlighter/dist/esm/styles/prism';
import mapPrismaEnumToLibraryLang from '../../utils/dataTransforms/mapPrismaEnumToLibraryLang';
import { Language } from '@prisma/client';
import classes from './CodeBlock.module.css';
import { IconCopy } from '@tabler/icons-react';
import { useEffect, useState } from 'react';

type Props = {
  language: Language;
  code: string;
};

export default function CodeBlock({ language, code }: Props) {
  const [isClicked, setIsClicked] = useState(false);

  const handleTooltipClick = () => {
    navigator.clipboard.writeText(code);
    setIsClicked(true);
  };

  useEffect(() => {
    if (!isClicked) return;

    const timeout = window.setTimeout(() => {
      setIsClicked(false);
    }, 1200);

    return () => {
      clearTimeout(timeout);
    };
  }, [isClicked]);

  return (
    <Box className={classes.wrapper}>
      <Tooltip label={isClicked ? 'Copied' : 'Copy'}>
        <ActionIcon
          onClick={handleTooltipClick}
          size="lg"
          variant={isClicked ? 'filled' : 'subtle'}
          className={classes.copy}
        >
          <IconCopy />
        </ActionIcon>
      </Tooltip>
      <Prism
        language={mapPrismaEnumToLibraryLang(language)}
        style={atomDark}
        showLineNumbers
        customStyle={{
          border: '1px solid gray',
        }}
      >
        {code}
      </Prism>
    </Box>
  );
}
