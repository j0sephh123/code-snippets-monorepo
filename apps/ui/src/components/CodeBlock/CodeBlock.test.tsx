import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import CodeBlock from './CodeBlock';

describe('components > CodeBlock', () => {
  it('renders', () => {
    const { asFragment } = render(
      <CodeBlock code="code" language="TypeScript" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
