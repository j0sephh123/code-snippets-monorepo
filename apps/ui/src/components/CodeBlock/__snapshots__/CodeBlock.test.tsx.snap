// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > CodeBlock > renders 1`] = `
<DocumentFragment>
  <pre
    style="color: rgb(197, 200, 198); text-shadow: 0 1px rgba(0, 0, 0, 0.3); font-family: Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: .5em 0px; overflow: auto; border-radius: 0.3em; background: rgb(29, 31, 33); border: 1px solid gray;"
  >
    <code
      class="language-typescript"
      style="color: rgb(197, 200, 198); text-shadow: 0 1px rgba(0, 0, 0, 0.3); font-family: Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 4; hyphens: none;"
    >
      <span>
        <span
          class="linenumber react-syntax-highlighter-line-number"
          style="display: inline-block; min-width: 1.25em; padding-right: 1em; text-align: right; user-select: none; color: rgb(124, 124, 124);"
        >
          1
        </span>
        <span>
          code
        </span>
      </span>
    </code>
  </pre>
</DocumentFragment>
`;
